source/common/stream_info/stream_info_impl.h

```diff
@@ -118,7 +120,9 @@ struct StreamInfoImpl : public StreamInfo {
   }
 
   void setResponseCodeDetails(absl::string_view rc_details) override {
-    response_code_details_.emplace(rc_details);
+    const absl::flat_hash_map<std::string, std::string> replacement{
```

Comment:
Drive by perf nit: it would probably be good to make this a construct on first use static to avoid making the map each time this is called? cc @alyssawilk

Reply:
https://github.com/envoyproxy/envoy/pull/13453